opensrp-tasking/src/test/java/org/smartregister/tasking/util/IndicatorUtilsTest.java

```diff
@@ -0,0 +1,123 @@
+package org.smartregister.tasking.util;
+
+
+import net.sqlcipher.Cursor;
+import net.sqlcipher.MatrixCursor;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.smartregister.domain.Task;
+import org.smartregister.tasking.BaseUnitTest;
+import org.smartregister.tasking.model.IndicatorDetails;
+import org.smartregister.tasking.model.TaskDetails;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+
+/**
+ * Created by ndegwamartin on 2019-09-27.
+ */
+public class IndicatorUtilsTest extends BaseUnitTest {
+
+    private Map<String, Set<Task>> taskTestMap = new HashMap<>();
+
+//    @Mock
+//    private SQLiteDatabase sqLiteDatabase;
+
+    @Before
+    public void setUp() {
+
+        TestingUtils.createTasks().forEach(t -> {
+            if (taskTestMap.containsKey(t.getStructureId())) {
+                taskTestMap.get(t.getStructureId()).add(t);
+            } else {
+                Set<Task> tasks = new HashSet<>();
+                tasks.add(t);
+                taskTestMap.put(t.getStructureId(), tasks);
+            }
+        });
+    }
+
+
+    @Test
+    public void validateSetupForTaskTestMapCorrectly() {
+        assertNotNull(taskTestMap);
+        assertTrue(taskTestMap.size() > 0);
+    }
+
+    @Test
+    public void testProcessTaskDetailsPopulatesTaskDetailsListCorrectly() {
+
+        List<TaskDetails> taskDetailsList = IndicatorUtils.processTaskDetails(taskTestMap);
+
+        assertNotNull(taskDetailsList);
+        assertTrue(taskDetailsList.size() > 0);
+        assertEquals(TestingUtils.createTasks().size(), taskDetailsList.size());
+
+    }
+
+    @Test
+    public void testProcessIndicatorsReturnsCorrectIndicatorDetails() {
+
+        List<TaskDetails> taskDetailsList = IndicatorUtils.processTaskDetails(taskTestMap);
+        IndicatorDetails indicatorDetails = IndicatorUtils.processIndicators(taskDetailsList);
+
+        assertNotNull(indicatorDetails);
+
+        assertEquals(3, indicatorDetails.getSprayed());
+        assertEquals(1, indicatorDetails.getNotSprayed());
+        assertEquals(2, indicatorDetails.getNotVisited());
+        assertEquals(50, indicatorDetails.getProgress());
+        assertEquals(6, indicatorDetails.getTotalStructures());
+        assertEquals(1, indicatorDetails.getIneligible());
+
+    }
+
+//    @Test
+//    public void testPopulateSprayIndicatorsPopulatesCorrectly() {
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix